### ISSUE TITLE
Uncaught TypeError: Cannot read property 'setValues' of undefined

### ISSUE THREAD

**Issue by matteox** [mulesoft/api-workbench#59]
[Enter steps to reproduce below:]

1. ...Create New Resource
2. ...Under get Method create New Response (200)
3 ...Attmpt to add annotation to response

screen:

  get:
    responses:
      200:
        annotations: 
        annotations: 
        annotations:
        annotations: 
        annotations: 
        annotations: 
        annotations:

**Atom Version**: 1.2.0
**System**: W7
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.6

### Stack Trace

Uncaught TypeError: Cannot read property 'setValues' of undefined

```
At C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:770

TypeError: Cannot read property 'setValues' of undefined
    at t.fromEditorToModel (C:\Users\Matthew\.atom\packages\api-workbench\main.js:30:3739)
    at C:\Users\Matthew\.atom\packages\api-workbench\main.js:30:2599
    at C:\Users\Matthew\.atom\packages\api-workbench\main.js:26:407
    at Array.forEach (native)
    at e.set (C:\Users\Matthew\.atom\packages\api-workbench\main.js:26:380)
    at t.e.setAssociatedValue (C:\Users\Matthew\.atom\packages\api-workbench\main.js:26:2934)
    at Array.n.emitter.handlersByEventName.did-change (C:\Users\Matthew\.atom\packages\api-workbench\main.js:27:6637)
    at Array.n.emitter.handlersByEventName.did-change (C:\Users\Matthew\.atom\packages\api-workbench\main.js:27:6631)
    at Array.n.emitter.handlersByEventName.did-change (C:\Users\Matthew\.atom\packages\api-workbench\main.js:27:6631)
    at n.emitter.handlersByEventName.did-change (C:\Users\Matthew\.atom\packages\api-workbench\main.js:27:6631)
    at Emitter.module.exports.Emitter.emit (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:86:11)
    at C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:211:32
    at Emitter.module.exports.Emitter.emit (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:86:11)
    at DisplayBuffer.module.exports.DisplayBuffer.emitDidChange (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\display-buffer.js:312:20)
    at DisplayBuffer.module.exports.DisplayBuffer.updateScreenLines (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\display-buffer.js:1251:19)
    at DisplayBuffer.module.exports.DisplayBuffer.handleTokenizedBufferChange (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\display-buffer.js:1213:19)
    at C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\display-buffer.js:5:61
    at Emitter.module.exports.Emitter.emit (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:86:11)
    at TokenizedBuffer.module.exports.TokenizedBuffer.handleBufferChange (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\tokenized-buffer.js:393:27)
    at C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\tokenized-buffer.js:69:24
    at Emitter.module.exports.Emitter.emit (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:86:11)
    at TextBuffer.module.exports.TextBuffer.applyChange (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:645:20)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:550:12)
    at Selection.module.exports.Selection.insertText (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\selection.js:482:43)
    at C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:732:29
    at C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:771:29
    at TextBuffer.module.exports.TextBuffer.transact (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:765:18)
    at TextEditor.module.exports.TextEditor.transact (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:1152:26)
    at C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:765:24
    at TextEditor.module.exports.TextEditor.mergeSelections (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:2017:18)
    at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:1979:35)
    at TextEditor.module.exports.TextEditor.mutateSelectedText (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:763:19)
    at TextEditor.module.exports.TextEditor.insertText (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor.js:729:19)
    at TextEditorComponent.module.exports.TextEditorComponent.onTextInput (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor-component.js:452:35)
    at HTMLDivElement.<anonymous> (C:\Users\Matthew\AppData\Local\atom\app-1.2.0\resources\app.asar\src\text-editor-component.js:3:61)
```

### Commands

```
     -2:30.7.0 api-workbench:Add new resource (button.icon.btn.icon-git-merge.btn-info.btn-sm)
     -2:21.8.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -1:11.2.0 core:undo (div.scrollpad.pane-item.padded.raml-details)
  3x -1:08.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:35.3.0 api-workbench:Create new Response (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  },
  "api-workbench": {}
}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.6
color-picker, v2.0.13
linter, v1.11.1
minimap, v4.16.0
raml, v0.2.0

# Dev
No dev packages
```

**Comment by ddenisenko** [comment]
@matteox Thanks for posting. Can not reproduce as-is, could you please provide a complete RAML file contents?

Also, please see the annotations section in the spec: http://docs.raml.org/specs/1.0/#raml-10-spec-annotations , there is no more "annotations" keyword in RAML, annotations are applied via the following syntax:

```RAML
/users:
  (testHarness): usersTest
  (badge): tested.gif
  (clearanceLevel):
    level: high
    signature: 230-ghtwvfrs1itr
  get:
    (experimental):
    (feedbackRequested):
    responses:
      200:
        (assertion): Condition 1
        (assertion): Condition 2
```
Please try this new one and let us know if it works for you.

**Comment by matteox** [comment]
Just to follow up - I imagine that the error comes from trying  to set values on "response 200" before it has been created.  I was typing in an annotation under the "create new response" dialog before saving the response.  I also did not declare annotations at the root..  Anyway - not an impediment.

**Comment by ddenisenko** [comment]
@matteox
> I was typing in an annotation under the "create new response" dialog before saving the response. 

This is the part, which make it reproducible, thank you!

So, the steps are the following:
1) Create RAML file with the following contents:
```RAML
#%RAML 1.0
title: Test

/test:
  get:

```

2) Put cursor at `get:` and activate `Create New Response` action from `Palette` in `Details` view.

3) Start typing annotation into `annotations` text field.

Result: Exception:
```
TypeError: Cannot read property 'setValues' of undefined
    at SimpleMultiEditor.fromEditorToModel (/Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/details2.js:503:55)
    at /Users/munch/work/apiworkbench/api-workbench/src/atom/raml1/details2.js:442:19
    at /Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:85:65
    at Array.forEach (native)
    at BasicBinding.set (/Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:85:28)
    at AtomEditorElement.BasicComponent.setAssociatedValue (/Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:201:27)
    at Array.AtomEditorElement.customize.editor.emitter.handlersByEventName.did-change (/Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:2270:23)
    at Array.AtomEditorElement.customize.editor.emitter.handlersByEventName.did-change (/Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:2269:22)
    at Array.AtomEditorElement.customize.editor.emitter.handlersByEventName.did-change (/Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:2269:22)
    at AtomEditorElement.customize.editor.emitter.handlersByEventName.did-change (/Users/munch/work/apiworkbench/api-workbench/src/atom/ramlscript/UI.js:2269:22)
    at Emitter.module.exports.Emitter.emit (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at /Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:211:32
    at Emitter.module.exports.Emitter.emit (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at DisplayBuffer.module.exports.DisplayBuffer.emitDidChange (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:312:20)
    at DisplayBuffer.module.exports.DisplayBuffer.updateScreenLines (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:1251:19)
    at DisplayBuffer.module.exports.DisplayBuffer.handleTokenizedBufferChange (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:1213:19)
    at __bind (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:5:61)
    at Emitter.module.exports.Emitter.emit (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at TokenizedBuffer.module.exports.TokenizedBuffer.handleBufferChange (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/tokenized-buffer.js:393:27)
    at /Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/tokenized-buffer.js:69:24
    at Emitter.module.exports.Emitter.emit (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at TextBuffer.module.exports.TextBuffer.applyChange (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:645:20)
    at TextBuffer.module.exports.TextBuffer.setTextInRange (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:550:12)
    at Selection.module.exports.Selection.insertText (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/selection.js:482:43)
    at /Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:732:29
    at /Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:771:29
    at TextBuffer.module.exports.TextBuffer.transact (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:765:18)
    at TextEditor.module.exports.TextEditor.transact (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1152:26)
    at /Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:765:24
    at TextEditor.module.exports.TextEditor.mergeSelections (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:2017:18)
    at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1979:35)
    at TextEditor.module.exports.TextEditor.mutateSelectedText (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:763:19)
    at TextEditor.module.exports.TextEditor.insertText (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor.js:729:19)
    at TextEditorComponent.module.exports.TextEditorComponent.onTextInput (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:452:35)
    at HTMLDivElement.__bind (/Users/munch/Downloads/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:3:61)
```

**Comment by ddenisenko** [comment]
Adding HIGH status as this is probably going to happen for all our annotation editing parts in Palette dialogs.

**Comment by ddenisenko** [comment]
Fixed. Fix should be available in 0.8.7.

Typing annotations works fine, annotations editing is removed from the dialog as the current editing system does not support it, and annotations were not supposed to appear in the editor.
